### DISCUSSION TITLE
How do you add a file/module to Android App Bundle (AAB) Modules in MAUI?

### DISCUSSION THREAD

**Discussion by TommiGustafsson-HMP** [dotnet/maui#14474] [Upgrading from Xamarin]
In Xamarin.Forms/Xamarin.Android, you can add files to Android App Bundle Modules using MSBuild in Xamarin.Android project file like this:

```
<ItemGroup>
    <AndroidAppBundleModules Include="assetpack\assets.zip" />
</ItemGroup>
```

It seems to be changed in MAUI. At least I could not find any reference to AndroidAppBundleModules in the MAUI repository. How are you supposed to add modules to the AAB in MAUI?

**Comment by xtuzy** [comment]
Xamarin.Android is dotnet-android, maui use dotnet-android, you can try use it in pure dotnet-android class library and reference this class library in maui app

**Reply by TommiGustafsson-HMP** [reply]
This is not helpful, since we are talking about MSBuild. What are you talking about?

**Comment by dellis1972** [comment]
@TommiGustafsson-HMP 

Maui uses the dotnet-android SDK underthe hood. So you should still be able to add `AndroidAppBundleModules` Items to your csproj as normal. Other SDKs (iOS, Windows etc) should ignore this ItemGroup since they don't use it. But the dotnet-android sdk will still pick them up. Just make sure your `AndroidPackageFormat` is set to `aab` in the csproj too.

**Reply by TommiGustafsson-HMP** [reply]
Thanks alot!
